File: Common/Entity/src/main/java/it/ey/dto/OVPStrategiaDTO.java
package it.ey.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)

public class OVPStrategiaDTO extends BaseDTO{
    private Long id;

    private String codStrategia;

    private String denominazioneStrategia;

    private String descrizioneStrategia;

    private String soggettoResponsabile;

    private List<OVPStrategiaIndicatoreDTO> indicatori;

    @JsonIgnore
    private OVPDTO ovp;

}

File: dfp-piao-fe/src/app/shared/models/classes/ovp-strategia-dto.ts
import { CampiTecniciDTO } from './campi-tecnici-dto';
import { OVPStrategiaIndicatoreDTO } from './ovp-strategia-indicatore-dto';

export class OVPStrategiaDTO extends CampiTecniciDTO {
  id?: number;
  codStrategia?: string;
  denominazioneStrategia?: string;
  descrizioneStrategia?: string;
  soggettoResponsabile?: string;
  indicatori?: OVPStrategiaIndicatoreDTO[];
}
